api_app/analyzers_manager/file_analyzers/yara_scan.py

```diff
@@ -115,23 +115,18 @@ def yara_update_repos():
         found_yara_dirs = []
         for analyzer_name, ac in analyzer_config.items():
             if analyzer_name.startswith("Yara_Scan"):
-                if analyzer_name == "Yara_Scan_YARAify_Rules":
+                yara_dirs = ac.param_values.get("git_repo_main_dir", [])
+                if not yara_dirs:
                     yara_dirs = ac.param_values.get("directories_with_rules", [])
-                    found_yara_dirs.extend(yara_dirs)
-                    yara_urls = ac.param_values.get("url", [])
+                yara_urls = ac.param_values.get("url", [])
+                if yara_urls:
+                    dir = "/opt/deploy/yara/yaraify_rules/rules"
```

Comment:
You are still forcing the path of the directory. If, for any reason, more than one analyzer use the `url` keyword the second analyzer will overwrite the rules of the first. 
Use `analyzer_name` inside the path to make it analyzer-unique.